runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/MetricsToPerStepNamespaceMetricsConverter.java

```diff
@@ -86,6 +87,38 @@ private static Optional<MetricValue> convertCounterToMetricValue(
                     .setValueInt64(value));
   }
 
+  /**
+   * @param metricName The {@link MetricName} that represents this counter.
+   * @param value The counter value.
+   * @return If the conversion succeeds, {@code MetricValue} that represents this counter. Otherwise
+   *     returns an empty optional
+   */
+  private static Optional<MetricValue> convertGaugeToMetricValue(
+      MetricName metricName,
+      Long value,
+      Map<MetricName, LabeledMetricNameUtils.ParsedMetricName> parsedPerWorkerMetricsCache) {
+
+    if ((!metricName.getNamespace().equals(BigQuerySinkMetrics.METRICS_NAMESPACE)
```

Comment:
We probably shouldn't do this check here, and separate out the check from the conversion itself. To be clear, this is checking that the namespace is either BQ or Kafka, right?

Reply:
or check that the namespace is in the set of supported namespaces

Reply:
Correct, if its not in that namespace, just skip the conversion.  Since we do that for all the metrics here, I moved the check out from the gauge and other metric converters.